sapio-data-repr/src/lib.rs

```diff
@@ -1,4 +1,39 @@
-use serde_json::Value;
+use serde::{self, Deserialize, Serialize};
+use serde_json::{Serializer, Value};
 
-pub struct SapioDataRepr(Value);
-pub struct SapioArgSchema(Value);
+#[derive(Debug)]
+pub struct Error(serde_json::Error);
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        todo!()
+    }
+}
+impl std::error::Error for Error {}
+
+struct SapioModuleBoundarySerializer<W> {
+    inner: Serializer<W>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct SapioModuleBoundaryRepr(Value);
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct SapioModuleSchema(Value);
+
+pub trait HasSapioModuleSchema {}
+
+pub fn to_string<T: Serialize>(v: &T) -> Result<String, Error> {
```

Comment:
you can define these as methods on a trait such as 

```
trait SapioDataReprExt {
   // ...
}

impl<T: Serialize> SapioDataReprExt for T {
 // ....
}

```

and then you will be able to call these methods on the type in question (&self receiver) instead of as free functions.

Reply:
Is that preferred? I mostly mimic'ed the API design of serde_json.

Reply:
up to you, fine as is too.

Having a trait can be helpful if you ever want to write generic functions that depend on these functions being avaialable in particular.

if you do it as a Ext trait it can sometimes also be helpful for e.g. autocompletion in IDEs.